src/metabase/pulse.clj

```diff
@@ -46,7 +46,7 @@
 (defn create-and-upload-slack-attachments!
   "Create an attachment in Slack for a given Card by rendering its result into an image and uploading it."
   [card-results]
-  (when-let [{channel-id :id} (slack/get-or-create-files-channel!)]
+  (when-let [{channel-id :id} (slack/files-channel)]
```

Comment:
Since `files-channel` is guaranteed to either return the channel or throw an Exception there's no point in using `when-let` here. Just use `let`.